supervision/dataset/formats/coco.py

```diff
@@ -0,0 +1,260 @@
+    approximation_percentage: float = 0.75,
+) -> Tuple[List[Dict], int]:
+    annotations_data = []
+    for xyxy, mask, confidence, class_id, tracker_id in detections:
+        polygon = []
+        if mask is not None:
+            polygon = list(
+                approximate_mask_with_polygons(
+                    mask=mask,
+                    min_image_area_percentage=min_image_area_percentage,
+                    max_image_area_percentage=max_image_area_percentage,
+                    approximation_percentage=approximation_percentage,
+                )[0].flatten()
+            )
+
+        per_label_dict = {}
+        per_label_dict["id"] = label_id
+        per_label_dict["image_id"] = image_id
+        per_label_dict["category_id"] = int(class_id)
+        box_width, box_height = xyxy[2] - xyxy[0], xyxy[3] - xyxy[1]
+        per_label_dict["bbox"] = [xyxy[0], xyxy[1], box_width, box_height]
+        per_label_dict["area"] = box_width * box_height
+        per_label_dict["segmentation"] = polygon
+        per_label_dict["iscrowd"] = 0
+        annotations_data.append(per_label_dict)
+        label_id += 1
+    return annotations_data, label_id
+
+
+def classes_to_coco_category_map(classes: List[str]) -> List[Dict]:
+    categories = []
+    for class_id, class_name in enumerate(classes):
+        cate_dict = {
+            "id": class_id,
+            "name": class_name,
+            "supercategory": "common-objects",
+        }
+        categories.append(cate_dict)
+    return categories
+
+
+def load_coco_annotations(
+    images_directory_path: str,
+    annotations_path: str,
+    force_masks: bool = False,
+) -> Tuple[List[str], Dict[str, np.ndarray], Dict[str, Detections]]:
+    """
+    Loads YOLO annotations and returns class names, images, and their corresponding detections.
+
+    Args:
+        images_directory_path (str): The path to the directory containing the images.
+        annotations_path (str): The path to the coco json annotation file.
+        force_masks (bool, optional): If True, forces masks to be loaded for all annotations, regardless of whether they are present.
+
+    Returns:
+        Tuple[List[str], Dict[str, np.ndarray], Dict[str, Detections]]: A tuple containing a list of class names, a dictionary with image names as keys and images as values, and a dictionary with image names as keys and corresponding Detections instances as values.
+    """
+    annotation_data = read_json_file(file_path=annotations_path)
+
+    classes = _extract_class_names(annotation_data=annotation_data)
+    images_infos = _extract_image_info(annotation_data=annotation_data)
+    image_id_label_id_pair, annotation_dict = _annotations_dict(
+        annotation_data=annotation_data
+    )
+
+    images = {}
+    annotations = {}
+
+    for images_info in images_infos:
+        image_path = os.path.join(images_directory_path, images_info["file_name"])
+        image = cv2.imread(str(image_path))
```

Comment:
Just tested dataset loading and noticed that images were not loaded correctly.

Here is my code:

```python
import roboflow
from roboflow import Roboflow

roboflow.login()

rf = Roboflow()
project = rf.workspace("roboflow-jvuqo").project("fashion-assistant-segmentation")
dataset = project.version(5).download("coco")

import supervision as sv

ds = sv.DetectionDataset.from_coco(
     images_directory_path=f"{dataset.location}/train/images",
     annotations_path=f"{dataset.location}/train/_annotations.coco.json",
)
```

And images got loaded as:

```
...
'photo_2023-05-30-17-34-54_jpeg.rf.593fa7603782b4679c624d5c1f63a508.jpg': None,
 'photo_2023-05-29-12-28-55_jpeg_jpg.rf.5a577cabb1aa09018e06fe01640bbb03.jpg': None,
 'photo_2023-05-29-12-29-08_jpeg_jpg.rf.5bd8bfb57cd46e45f2fd8ff9bfa34de1.jpg': None,
 'photo_2023-05-29-14-10-23_jpeg_jpg.rf.5d4449e52e9dcdf25ef1f5b3f1c1b62c.jpg': None,
 'photo_2023-05-30-17-35-05-1-_jpeg.rf.5c217edf22eab45f1761f3972453d469.jpg': None,
 'photo_2023-05-30-17-34-57_jpeg.rf.5959091a468b991b5794778f093c881a.jpg': None,
 'photo_2023-05-29-12-29-12_jpeg_jpg.rf.5b0744781d8f14023a55a9edadfc23b1.jpg': None,
...
```